src/nm.c

```diff
@@ -365,6 +365,7 @@ write_wireguard_params(const NetplanNetDefinition* def, GKeyFile *kf, GError** e
         if (peer->preshared_key) {
             if (peer->preshared_key[0] == '/' && !is_wireguard_key(peer->preshared_key)) {
                 g_set_error(error, G_MARKUP_ERROR, G_MARKUP_ERROR_INVALID_CONTENT, "%s: shared key needs to be base64 encoded when using the NM backend\n", def->id);
+                g_free(tmp_group);
```

Comment:
This is fine!

suggestion: How about using `g_autofree gchar* tmp_group = NULL;` in the definition above, instead of this? That way we wouldn't need to take as much care in the future.

Reply:
Yeah, that works. I will just need to move the variable declaration to inside the for loop. Because it will be overridden with a new allocation for each iteration.